mlflow/genai/judges/instructions_judge/constants.py

```diff
@@ -0,0 +1,23 @@
+"""Constants for InstructionsJudge."""
+
+INSTRUCTIONS_JUDGE_EVALUATION_PROMPT_TEMPLATE = """
```

Comment:
How did we arrive at this prompt? 

Why I ask: I believe that DSPy would form a much simpler prompt for a similar signature and task description. This prompt is much more complicated so I wonder if we can simplify to just {{task_instructions}} with additional substitutions for the variables.

Reply:
Here is an alternative proposal:

- We use `instructions` as the system prompt, perhaps minimally wrapped as "You are a helpful judge. Your task: {{instructions}}. Generate output in the following format: {{format}}"
- We do variable substitution in the `user` message

I believe this mirrors what happens in `optimize_prompt` and it's also a good way to lower to/lift from dspy for the actual alignment. 

Thoughts?

Reply:
This definitely helps, but the other part of @alkispoly-db's comment about variable formatting still needs to be addressed - https://github.com/mlflow/mlflow/pull/17466/files#r2317059803